mlflow/server/js/src/experiment-tracking/components/experiment-page/hooks/useRunsArtifacts.test.tsx

```diff
@@ -0,0 +1,59 @@
+import { cleanup, waitFor } from '@testing-library/react';
+import { listArtifactsApi } from '../../../actions';
+import { useRunsArtifacts } from './useRunsArtifacts';
+import { ArtifactListFilesResponse } from '../../../types';
+import { renderHook } from '@testing-library/react-for-react-18';
+
+const mockArtifactsData: Record<string, ArtifactListFilesResponse> = {
+  'run-1': {
+    root_uri: 'run-1',
+    files: [
+      {
+        path: 'artifact1.txt',
+        is_dir: false,
+        file_size: 300,
+      },
+    ],
+  },
+  'run-2': {
+    root_uri: 'run-2',
+    files: [
+      {
+        path: 'artifact2.txt',
+        is_dir: false,
+        file_size: 300,
+      },
+    ],
+  },
+};
+
+jest.mock('../../../actions', () => ({
+  ...jest.requireActual('../../../actions'),
+  listArtifactsApi: jest.fn((runUuid: string) => {
+    return {
+      payload: mockArtifactsData[runUuid],
+    };
+  }),
+}));
+
+describe('useRunsArtifacts', () => {
+  afterEach(() => {
+    jest.restoreAllMocks();
+    cleanup();
+  });
+
+  test('fetches artifacts for given run UUIDs', async () => {
+    const { result } = renderHook(() => useRunsArtifacts(['run-1', 'run-2']));
```

Comment:
the issue here is that creating the array inline with `['run-1', 'run-2']` causes the hook to constantly rerender, as the `runUuids` is not referentially equivalent (a new array is created every time)

```suggestion
    const runUuids = ['run-1', 'run-2'];
    const { result } = renderHook(() => useRunsArtifacts(runUuids));
```